wordpress-mobile/gutenberg-mobile
# Accessibility: Handle the iOS z-gesture to exit modals and block selection

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/15153

This PR improves accessibility on modals and blocks. It also makes images easily editable by listening for a long press event on the image.

### Testing Instructions (iOS)
- Activate VoiceOver (Settings > General > Accessibility > VoiceOver) or say "Hey Siri, activate VoiceOver"
- Boot the demo app on iOS, go from block to block by swiping on the right
- Notice the screen reader says "Image block. Empty. Button. Double tap to select an image" for image blocks without an image
- Notice the screen reader says "Image block. Button" for non empty image block
- Double tap on a selected non-empty image block
- Notice it says "Image block. Long press to edit the image. button image."
- Go to the next item and notice it says "Image caption. Empty. Button. Actions available"
- Go back (swipe left) to the image and notice it says "Long press to edit the image. button image."
- Long press the image (when VoiceOver is active you may have to tap once first, it's like a double tap but on the second tap your finger stays on the screen, you will hear 3 sounds in this case) until the screen to edit the image pops up. Edit the image if you want to
- Do the z-gesture (make a z with 2 figers on the screen) and notice the currently selected block is unselected
- Try to close other modals with the z-gesture

### Testing Instructions (Android)
- Activate TalkBack (Settings > Accessibility > TalkBack)
- Boot the demo app on Android, go from block to block by swiping on the right
- Run the same tests as on iOS (ignore the fact that the block container is accessible at the moment)
- Notice that by swiping right you enter inside the block (RN ignores the fact that an element is a child of an accessible element)
- Notice the z-gesture does not work for unselecting a block
- Notice that the double long tap works to edit the image but works for modals
